File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/Basic/RemotingEvents/EventRegistration/EventRegistration.cs
/*=====================================================================

  File:      EventRegistration.cs

---------------------------------------------------------------------
This file is part of the Microsoft .NET Framework SDK Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

This source code is intended only as a supplement to Microsoft
Development Tools and/or on-line documentation.  See these other
materials for detailed information regarding Microsoft code samples.
 
THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.
=====================================================================*/

using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels;
using System.Runtime.Remoting.Channels.Http;

using Zap;
using Wak;

public class Client
{
    public static void Main(String[] args)
    {
        // This could also be done with a Remoting configuration file

        // Register the HTTP Channel
        ChannelServices.RegisterChannel(new HttpChannel(0));

        Baz baz = (Baz)Activator.GetObject(
                                          typeof(Baz),
                                          "http://localhost:888/EventSinkHost/Baz.soap"
                                          );

        // Obtain a Proxy to the SOAP URL
        Waz waz = (Waz)Activator.GetObject(
                                          typeof(Waz),
                                          "http://localhost:999/EventFireHost/Waz.soap"
                                          );

        // Subscribe to event : occurs over SOAP
        waz.Greeting += new GreetingEvent(baz.GreetingHandler);

        waz.HelloMethod("Hi from the client");
                   
        // Unsubscribe to event : occurs over SOAP
        waz.Greeting -=new GreetingEvent(baz.GreetingHandler); 
    }
}

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/VB/Basic/RemotingEvents/EventRegistration/EventRegistration.vb

Imports System
Imports System.Runtime.Remoting
Imports System.Runtime.Remoting.Channels
Imports System.Runtime.Remoting.Channels.Http

Imports Zap
Imports Wak



Public Class Client
    
    Public Shared Sub Main()
        ' This could also be done with a Remoting configuration file
        ' Register the HTTP Channel
        ChannelServices.RegisterChannel(New HttpChannel(0))
        
        Dim baz As Baz = CType(Activator.GetObject(GetType(Baz), "http://localhost:888/EventSinkHost/Baz.soap"), Baz)
        
        ' Obtain a Proxy to the SOAP URL
        Dim waz As Waz = CType(Activator.GetObject(GetType(Waz), "http://localhost:999/EventFireHost/Waz.soap"), Waz)
        
        ' Subscribe to event : occurs over SOAP
        AddHandler waz.Greeting, AddressOf baz.GreetingHandler
        
        waz.HelloMethod("Hi from the client")
        
        ' Unsubscribe to event : occurs over SOAP
        RemoveHandler waz.Greeting, AddressOf baz.GreetingHandler
    End Sub 'Main
End Class 'Client